### ISSUE TITLE
Anonymisation and assumption of invariant in wrong order

### ISSUE THREAD

**Issue by mattulbrich** [JonasKlamroth/JJBMC#2]
Refer to the following example:
````java
    void n() {
        int[] array = new int[1];
        int t = 0;
        //@ loop_invariant t == 0;
        //@ loop_modifies array[t];
        while(true) {
            t = 0;
            mod(array, t);
        }
    }
````

On the anonymisation of the loop_modifies. array[t] is renewed. However, t==0 is only assumed afterwards. Kind of too late to help.

````java
    int t = 0;
      {
        {
          assert t == 0;
        }
      }
      t = CProver.nondetInt();
      if (array != null) {
        for (int __tmpVar__0 = t; __tmpVar__0 <= t; ++__tmpVar__0) {
          array[__tmpVar__0] = CProver.nondetInt();
        }
      }
      {
        CProver.assume(t == 0);
      }
````

**Comment by JonasKlamroth** [comment]
Thats right. However i have still not found an easy way determine the order in which the havoc and assumption statements have to occure. Assuming before havocing wont do the trick either so it has to be some kind of interweaving. However this seems to be pretty complex. do you have any suggestions on how this could be done?

**Comment by JonasKlamroth** [comment]
fixed by 2d213c04457718aa5e119c6e7fa01fdfaed37a8b
